### PR TITLE
smack: remove "User" domain

### PR DESCRIPTION
**Pull request by pohly**
The entire "User" domain support has been tentative. It depends a lot
on the actual application framework and thus is neither necessarily
the same everywhere nor absolutely needed. For example, Ostro OS does
not need it.

Support for it also was incomplete, probably even in Tizen, leading to
problems even when a distro has no use for the "User" domain: in Ostro
OS, the PAM plugin sets the process label to "User" even when logging
in as root, and then when "passwd" is used to modify /etc/passwd, the
new file has "User" as Smack label because there is no transmute Smack
rule for User -> System::Shared and thus the transmute flag on /etc
gets ignored.

Instead of trying to fix this, it is easier to simplify the entire
layers by removing the "User" labels and rules. What remains is
hard-coded in the upstream recipes.

If someone need support for a "User" domain, they can add that to
their custom distro together with the code that depends on it or (when
that is not easy, for example in the smacknet tool) propose patches
which add back what is needed in a way that a) it is properly
documented and b) optional.

@jobol @ipuustin please review. I also need to do some more testing
myself.

### CODE REVIEW

**Comment by acornea** [comment]
If we plan on not using User domain anymore by default, maybe we should also remove that from meta-security-smack/recipes-core/base-files/base-files_%.bbappend

**Comment by jobol** [comment]
Thinking about this since yesterday...

I think that it's cool to have User context when logged. But if you don't need it....

From your comment I read that
1. it is not proven to be fully functional
2. it bother humans when editing files

Killing the User domain only for the second reason doesn't make sense.

But telling that nobody wants to work on that topic to make it work or that it works without it can be heard. What is your meaning @cschaufler?

**Comment by pohly** [comment]
On Thu, 2016-04-21 at 05:09 -0700, acornea wrote:

> If we plan on not using User domain anymore by default, maybe we
> should also remove that from
> meta-security-smack/recipes-core/base-files/base-files_%.bbappend

What is left from User there? The commit removes "User" also from that
file.

**Comment by acornea** [comment]
Apologies, missed that in between the many red lines :)

**Comment by pohly** [comment]
> From your comment I read that
> 
> ```
>   1. it is not proven to be fully functional
> 2. it bother humans when editing files
> ```

The problem which broke the camel's back is that calling simple commands
like "passwd" leads to unexpected results.

> Killing the User domain only for the second reason doesn't make sense.

It does. Why fix a problem in something that isn't even needed?

> But telling that nobody wants to work on that topic to make it work or
> that it works without it can be heard. What is your meaning
> @cschaufler?

Casey was the one who suggested that the "User" domain should be removed
because it is not used in Ostro.

I'll probably go one step further and also remove unused components like
Cynara and SecurityManager. They are just a maintenance burden at this
point, and if someone needs them, they can copy the recipes and continue
maintaining them elsewhere.

Update: the above has inline quotes that do not render correctly because I sent the comment via email - looks like a bug in the Github mail interface. I can't even fix it by editing the comment.

File: meta-security-smack/lib/oeqa/runtime/smack.py [hunk 1]
```diff
@@ -566,8 +566,8 @@ def test_smack_labels(self):
 /etc access="System::Shared" transmute="TRUE"
 /etc/passwd access="System::Shared"
 /etc/terminfo access="System::Shared" transmute="TRUE"
-/etc/skel access="User::Home"
-/etc/skel/.profile access="User"
```
Comment (jobol) [inline meta-security-smack/lib/oeqa/runtime/smack.py:570]: a bug in the previous code (no file object should be labelled User IIRC)

**Comment by jobol** [comment]
Okay. I would like to have a PAM module for smack that has the configuration to set a label to User or to anything else. It is possible I think. For the remaining, it is up to the distro to set its rules.

I think that the correct place for a such PAM module is in smack-userspace of @smack-team (but the maintainer is sleeping a lot)

**Comment by pohly** [comment]
@jobol if you need such a PAM plugin and can't find someone to maintain it for you, then you will have to maintain it yourself in your own layer.

**Comment by pohly** [comment]
retest this please

**Comment by pohly** [comment]
Force pushed a fixed "smack: remove "User" domain" patch (smack label test failed because now /etc/skel is no longer special and inherits the transmute flag).

Also added much better error output when the smack label test fails. Perhaps that'll help understand why /var/log ends up with a different label than expected (seen elsewhere, for example in https://ostroproject.org/jenkins/job/dsl-meta-ostro_pull-requests/295/).

**Pull request merged**